chirho/explainable/handlers/explanation.py

```diff
@@ -85,13 +91,20 @@ def SearchForExplanation(
     :param alternatives: An optional mapping of names to alternative antecedent interventions.
     :param factors: An optional mapping of names to consequent constraint factors.
     :param preemptions: An optional mapping of names to witness preemption values.
-    :param antecedent_bias: The scalar bias towards not intervening. Must be between -0.5 and 0.5, defaults to 0.0.
-    :param consequent_scale: The scale of the consequent factor functions, defaults to 1e-2.
-    :param witness_bias: The scalar bias towards not preempting. Must be between -0.5 and 0.5, defaults to 0.0.
+    :param antecedent_bias: A scalar bias towards not intervening. Must be between -0.5 and 0.5. Defaults to 0.0.
+    :param consequent_scale: The scale of the consequent factor functions. Defaults to 1e-2.
+    :param witness_bias: A scalar bias towards not preempting. Must be between -0.5 and 0.5. Defaults to 0.0.
     :param prefix: A prefix used for naming additional consequent nodes. Defaults to ``__consequent_``.
+    :param num_samples: The number of samples to be drawn for each antecedent and witness.
+                        Needed if witness and antecedent samples are to be coordinated.
```

Comment:
I can't tell from the code or this rather mysterious fragment of documentation what either of the parameters `num_samples` or `sampling_dim` are supposed to achieve or what effect different values are supposed to have on the behavior of `SearchForExplanation`, especially as the other parameters are varied.

Maybe that's on me, but since I know something about the inner workings of `SearchForExplanation`, I think it's reasonable to assume other people who don't will have an even harder time understanding. I'm also not sure this is a solvable problem with documentation alone - the interface for `SearchForExplanation` is already so complex that there's really no room to add more choice to it.